packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -56,35 +135,107 @@ export class AssetStaging extends Construct {
 
   private readonly relativePath?: string;
 
+  private readonly bundleDir?: string;
+
   constructor(scope: Construct, id: string, props: AssetStagingProps) {
     super(scope, id);
 
     this.sourcePath = props.sourcePath;
     this.fingerprintOptions = props;
-    this.sourceHash = FileSystem.fingerprint(this.sourcePath, props);
 
     const stagingDisabled = this.node.tryGetContext(cxapi.DISABLE_ASSET_STAGING_CONTEXT);
-    if (stagingDisabled) {
-      this.stagedPath = this.sourcePath;
+
+    if (props.bundling) {
+      // Create temporary directory for bundling
+      this.bundleDir = fs.mkdtempSync(path.resolve(path.join(os.tmpdir(), 'cdk-asset-bundle-')));
+
+      // Always mount input and output dir
+      const volumes = [
+        {
+          hostPath: this.sourcePath,
+          containerPath: BUNDLING_INPUT_DIR,
+        },
+        {
+          hostPath: this.bundleDir,
+          containerPath: BUNDLING_OUTPUT_DIR,
+        },
+        ...props.bundling.volumes ?? [],
```

Comment:
I don’t believe you need the `?? []` here.

Reply:
It is needed, those are the optional additional volumes to mount
https://github.com/aws/aws-cdk/pull/7898/files/587bc8235fbb2d1b1a33d8a22e2fe03495d47866#diff-e50eed711a8f4aa2a04094469a0cf89aR58

Reply:
Yes, you are right. Sorry.,